### ISSUE TITLE
Feature Request - Azure support

### ISSUE THREAD

**Issue by twalderman** [longy2k/obsidian-bmo-chatbot#46]
User Story:As a BMO Userr, I want to configure an Azure-hosted GPT connection within the BMO application settings so that we can leverage its capabilities for AI service providers. 

Acceptance Criteria:

1. Given that I am configuring the application settings, When I input the Azure connection details for hosted OAI Then the system validates the connection parameters and saves the configuration securely.

Support the following criteria: 
Endpoint field
API key field
Vaidate button

2. Given that a valid azure connection exists, provisioned models can be selected in the setting modal and via the /model command

**Comment by choigawoon** [comment]
i also need this

**Comment by longy2k** [comment]
I looked into adding support for Azure OpenAI but I currently do not have access to the API Key since I do not have a proper work account.

I can see that they have the documentation for Azure OpenAI available so I can probably work it out from there.

I will come back at a later date to see if I can add it.

**Comment by twalderman** [comment]
Ping me on email to discuss api key. When you are ready I can lend you mine for testing but it will need to be super quick.

**Comment by tomaspiaggio** [comment]
@longy2k I also need this and I'm willing to push a PR. Let me know if that helps!

**Comment by longy2k** [comment]
@tomaspiaggio Yes, if you can create a PR, I'll definitely push it through! Thanks

**Comment by tomaspiaggio** [comment]
@longy2k I don't know if you saw the PR I made #112. Waiting for any comments you may have!

**Comment by Lesaloon** [comment]
+1

**Comment by longy2k** [comment]
Sorry, I will get to it soon. I have some major life changes that I have to adjust to. Thank you!

**Comment by tomaspiaggio** [comment]
@Lesaloon if you need you can clone my code, build it and manually set it as the extension in the meantime. That's what I'm doing right now and it works great!

**Comment by Lesaloon** [comment]
@tomaspiaggio it works well, thank you !
